src/state/slices/assetsSlice/assetsSlice.ts

```diff
@@ -57,6 +109,11 @@ export const assets = createSlice({
       state.byId = { ...state.byId, ...action.payload.byId } // upsert
       state.ids = Array.from(new Set([...state.ids, ...action.payload.ids]))
     },
+    upsertAsset: (state, action: PayloadAction<Asset>) => {
```

Comment:
consumers can now upsert assets into the store at runtime from external data sources, e.g. monkey pics/LP tokens